src/main/java/seedu/address/model/task/Task.java

```diff
@@ -0,0 +1,150 @@
+
+    /**
+     * The first character must not be a whitespace,
+     * otherwise " " (a blank string) becomes a valid input.
+     */
+    public static final String VALIDATION_REGEX = "[^\\s].*";
+    private final String name;
+    private boolean isDone;
+    private Employee employee;
+    private final LocalDateTime deadline;
+    /**
+     * Constructs a {@code Task}
+     * @param taskName The name of the task.
+     */
+    public Task(String taskName, LocalDateTime deadline) {
+        requireNonNull(taskName);
+        requireNonNull(deadline);
+        name = taskName;
+        this.deadline = deadline;
+        this.isDone = false;
+    }
+
+    /**
+     * Constructs a {@code Task} with a pre-determined Done status - for loading of tasks from taskhub.json.
+     * @param taskName The name of the task.
+     */
+    public Task(String taskName, LocalDateTime deadline, Boolean isDone) {
+        requireNonNull(taskName);
+        requireNonNull(deadline);
+        name = taskName;
+        this.deadline = deadline;
+        this.isDone = isDone;
+    }
+
+    /**
+     * Returns true if a given string is a valid project name.
+     */
+    public static boolean isValidTask(String testString) {
+        return testString.matches(VALIDATION_REGEX);
+    }
+
+    /**
+     * Assigns an employee to the task.
+     */
+    public void setEmployee(Employee employee) {
+        this.employee = employee;
+    }
+
+
+    /**
+     * Tells us if a task is done.
+     * @return A boolean value indicating if the task is done.
+     */
+    public boolean isDone() {
+        return this.isDone;
+    }
+
+    /**
+     * Marks a task as done.
+     */
+    public void markAsDone() {
+        this.isDone = true;
+    }
+
+    /**
+     * Marks a task as not done.
+     */
+    public void markAsUndone() {
+        this.isDone = false;
+    }
+
+    public Employee getEmployee() {
+        return this.employee;
+    }
+
+    public String getName() {
+        return this.name;
+    }
+    public LocalDateTime getDeadline() {
+        return this.deadline;
+    }
+
+    @Override
+    public String toString() {
+        String completionString = this.isDone() ? "[X]" : "[]";
+        return completionString + " | "
+                                    + this.name + " | "
+                                        + this.employee + " | "
+                                            + DateTimeFormatter.ofPattern("yyyy-MM-dd HHmm").format(this.deadline);
+    }
+
+    /**
+     * Determines if a string is of the format "yyyy-MM-dd HHmm" i.e., a valid LocalDateTime object.
+     *
+     * @param input The string to be examined.
+     * @return A boolean value which tells us if the string represents a LocalDate.
+     */
+    public static boolean isValidDateTime(String input) {
+        Pattern pattern = Pattern.compile("^\\d{4}-\\d{2}-\\d{2} \\d{4}$");
+        Matcher matcher = pattern.matcher(input);
+        return matcher.matches();
+    }
+
```

Comment:
Actually the static `parse()` method of the `LocalDateTime` class automatically throws a `DateTimeParseException` if the format is invalid, so perhaps this method is not needed.